app/components/RecentEvents.tsx

```diff
@@ -67,6 +83,68 @@ export default class RecentEvents extends TsxComponent<{}> {
     } catch (e) {}
   }
 
+  setNative(native: boolean) {
+    if (!native && this.customizationService.state.eventsSize < 250) {
+      // Switch to a reasonably sized events feed
+      this.customizationService.setSettings({ eventsSize: 250 });
+    }
+
+    if (!native) {
+      this.dismissablesService.dismiss(EDismissable.RecentEventsHelpTip);
+    }
+
+    this.customizationService.setSettings({ legacyEvents: !native });
+  }
+
+  handleBrowserViewReady(view: Electron.BrowserView) {
+    if (view.isDestroyed()) return;
+
+    electron.ipcRenderer.send('webContents-preventPopup', view.webContents.id);
+
+    view.webContents.on('new-window', (e, url) => {
+      const match = url.match(/dashboard\/([^\/^\?]*)/);
+
+      if (match && match[1] === 'recent-events') {
+        this.popoutRecentEvents();
+      } else if (match) {
+        this.navigationService.navigate('Dashboard', {
+          subPage: match[1],
+        });
+      } else {
+        electron.remote.shell.openExternal(url);
+      }
+    });
+  }
+
+  renderNativeEvents(h: Function) {
+    return (
+      <div class={styles.eventContainer}>
+        {!!this.recentEvents.length &&
+          this.recentEvents.map(event => (
+            <EventCell
+              event={event}
+              repeatAlert={this.repeatAlert.bind(this)}
+              eventString={this.eventString.bind(this)}
+            />
+          ))}
+        {this.recentEvents.length === 0 && (
+          <div class={styles.empty}>{$t('There are no events to display')}</div>
+        )}
+      </div>
+    );
+  }
+
+  renderEmbeddedEvents(h: Function) {
+    return (
+      <BrowserView
+        class={styles.eventContainer}
+        src={this.userService.recentEventsUrl()}
+        setLocale={true}
```

Comment:
Don't see `setLocale` in BrowserView docs. How does it work? Does it take a locale from cookies?

Reply:
This is not the electron `BrowserView` directly, since `BrowserView` is not a DOM-based API like `webview`.  This is our custom Vue component that makes it behave more like a `webview`.  The `setLocale` code was ported from our `webview` implementation.